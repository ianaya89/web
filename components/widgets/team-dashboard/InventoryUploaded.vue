<template>
  <div class="w-full">
    <div class="w-full flex px-2">
      <div class="mr-auto">
        <label-chart :label="`${$get(unit,'entity.inventoryStatus.percentage',0)}%`" :chart-value="$get(unit,'entity.inventoryStatus.percentage',0) / 100" :info="$t('app.team.dashboard.inventory_uploaded')" />
      </div>
    </div>

    <div class="w-full bg-white shadow-xl px-4 py-8">
      <div class="icon-scroll">
        <icon-check v-for="item in $get(unit,'entity.inventoryStatus.allCheckedCategories',[])" :key="item.id" class="icon-check" :value="$get(unit,'entity.inventoryStatus.selectedCategories',[]).find(selectedCategoryItem => selectedCategoryItem.id ===item.id)!=null" :clickable="false">
          <inventory-icon class="w-full h-full" :category="item" />
        </icon-check>
      </div>

      <app-button class="w-full text-left" :title="$t('app.team.dashboard.upload_inventory')" :show-arrow="true" :btn-class="'primary-lg'" @clicked="onUploadInventoryClick()" />
    </div>
  </div>
</template>

<script>
import LabelChart from '@/components/LabelChart'
import { UnitInventoryStatusQuery } from '@/services/graphql'
import AppButton from '@/components/AppButton'
import IconCheck from '@/components/IconCheck'
import InventoryIcon from '@/components/inventory/InventoryIcon'

export default {
  name: 'InventoryUploaded',

  components: {
    InventoryIcon,
    IconCheck,
    AppButton,
    LabelChart
  },

  props: {
    unitId: {
      type: String,
      required: true
    }
  },

  apollo: {
    unit: {
      ...UnitInventoryStatusQuery,
      variables() {
        return {
          id: this.unitId
        }
      }
    }
  },

  methods: {
    onUploadInventoryClick() {
      this.$router.push(
        this.localePath({
          name: 'unit-id-inventory',
          params: {
            id: this.unitId
          }
        })
      )
    }
  }
}
</script>

<style lang="postcss" scoped>
.icon-scroll {
  @apply .mb-4 .overflow-x-auto .whitespace-no-wrap;
  -webkit-overflow-scrolling: touch;
  -ms-overflow-style: -ms-autohiding-scrollbar;

  .icon-check {
    @apply .p-2 .inline-block;
  }
}
</style>
