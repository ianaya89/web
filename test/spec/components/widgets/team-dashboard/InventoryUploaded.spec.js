import { shallowMount } from '@vue/test-utils'
import InventoryUploaded from '@/components/widgets/team-dashboard/InventoryUploaded'

describe('InventoryUploaded', () => {
  it('renders InventoryUploaded', () => {
    const wrapper = shallowMount(InventoryUploaded, {
      propsData: {
        unitId: 1
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
