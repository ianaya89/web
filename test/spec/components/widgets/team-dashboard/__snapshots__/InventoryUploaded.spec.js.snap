// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InventoryUploaded renders InventoryUploaded 1`] = `
<div
  class="w-full"
>
  <div
    class="w-full flex px-2"
  >
    <div
      class="mr-auto"
    >
      <label-chart-stub
        animation="true"
        chartvalue="0"
        info="TRANSLATE:app.team.dashboard.inventory_uploaded"
        label="0%"
        labelclass="text-h3"
        piecolors="#21CE99,#D4D4D4"
      />
    </div>
  </div>
   
  <div
    class="w-full bg-white shadow-xl px-4 py-8"
  >
    <div
      class="icon-scroll"
    />
     
    <app-button-stub
      btnclass="primary-lg"
      class="w-full text-left"
      showarrow="true"
      title="TRANSLATE:app.team.dashboard.upload_inventory"
    />
  </div>
</div>
`;
